packages/drivers/odsp-driver/src/graph.ts

```diff
@@ -26,6 +28,38 @@ function slashTerminatedOriginOrEmptyString(origin: string | undefined) {
         : "";
 }
 
+const getSPOAndGraphRequestIdsFromResponse = async (headers: Map<string, string>) => {
+    interface LoggingHeader {
+        headerName: string;
+        logName: string;
+    }
+    // We rename headers so that otel doesn't scrub them away. Otel doesn't allow
+    // certain characters in headers including '-'
+    const headersToLog: LoggingHeader[] = [
+        { headerName: "sprequestguid", logName: "spRequestGuid" },
+        { headerName: "request-id", logName: "requestId" },
+        { headerName: "client-request-id", logName: "clientRequestId" },
+        { headerName: "x-msedge-ref", logName: "xMsedgeRef" },
+    ];
+    const additionalProps: ITelemetryProperties = {};
+    if (headers) {
+        headersToLog.forEach((header) => {
+            const headerValue = headers.get(header.headerName);
+            if (headerValue) {
```

Comment:
Better check for null, as I'd assume we want to log empty strings

Reply:
In general, you can leverage null-> undefined conversion (I think I've added it for similar thing somewhere in driver) and not even check result, always setting property.

Reply:
As I checking in ts, headers.get() is returning string or undefined. So I don't think we need null to undefined conversion.

Reply:
Response.headers.get definetly returns string | null. Take a look at nullToUndefined and code in throwOdspNetworkError

Reply:
Yes but here it is Map and not Response.headers.